overlord/snapstate/snapstate.go

```diff
@@ -2357,6 +2362,11 @@ func AddGadgetAssetsTasks(st *state.State, ts *state.TaskSet) (*state.TaskSet, e
 	gadgetCmdline.Set("snap-setup-task", snapSetupTask.ID())
 	gadgetCmdline.WaitFor(gadgetUpdate)
 	ts.AddTask(gadgetCmdline)
+	// make sure that remodel can identify which tasks introduce actual
+	// changes to the system and order them correctly
+	if edgeTask := ts.MaybeEdge(DownloadAndChecksDoneEdge); edgeTask == nil {
+		ts.MarkEdge(allTasks[len(allTasks)-1], DownloadAndChecksDoneEdge)
+	}
```

Comment:
Similar question, can we use gadgetCmdline task instead of allTasks[len(allTasks)-1] ?